frontend/src/concepts/k8s/K8sNameDescriptionField/K8sNameDescriptionField.tsx

```diff
@@ -43,8 +51,15 @@ type K8sNameDescriptionFieldProps = {
   hideDescription?: boolean;
 };
 
+// using enum to make the type; cannot use enum directly
```

Comment:
There is already a pattern we use for adding validation to this specific component. You wont need to touch this file. instead follow ManageProjectModal for how it is done and replicate that for the clusterStorageModal